enhancements/network/routed-ingress-primary-udn-static-ips.md

```diff
@@ -0,0 +1,723 @@
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+
+# Routed ingress support for primary UDN attached VMs with static IPs
+
+## Summary
+
+There's an ongoing interest to migrate Virtual Machines (VMs) from other
+virtualization platforms into OpenShift. These users are both after a managed IP
+experience (which they have on their existing platform), and also want to
+consume existing Kubernetes features, like network policies, and sometimes
+services.
+
+Traditional virtualization users have some expectations on what a
+virtualization platform should provide. Live-migration, and IP address
+persistence across reboots are paramount features for any virtualization
+solution, and OpenShift Virtualization currently supports these features on its
+primary UserDefinedNetworks (UDNs).
+
+These users have additional requirements, like routed ingress into their VMs,
+and, to import the VMs with the MAC, IPs, and gateway configuration the VM had
+on the previous platform.
+
+Routed ingress into the VMs is being added to OpenShift already, and is tracked
+by [this enhancement](https://github.com/openshift/enhancements/pull/1636/files)
+which tracks adding BGP support to OpenShift.
+
+Given the aggressive time-frame we're attempting, adding support for importing
+a VM into OpenShift with it's MAC, IPs, and gateway on top of primary UDN is the
+missing piece we should pursue.
+
+## Motivation
+
+Some users are running VMs in virtualization platforms having a managed IP
+configuration.
+
+For this kind of users, we need to have a way to enable their existing VMs to
+run properly after being migrated into OpenShift, without any guest
+configuration changes. For that, we need to import the VMs from their existing
+platforms, preserving their existing MACs, IPs, and gateway configuration.
+
+### User Stories
+
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM - whose IPs were statically configured - into Kubernetes,
+attaching it to an overlay network. I want to ingress/egress using the same IP
+address (the one assgined to the VM).
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM into Kubernetes, attaching it to an overlay network. I want to
+have managed IPs for my VMs, since that feature was already available in my
+previous platform.
+- As the owner of a VM running in a traditional virtualization platform, I want
+to import said VM into Kubernetes, attaching it to an overlay network. I want to
+be able to consume Kubernetes features like network policies, and services, to
+benefit from the Kubernetes experience.
+
+### Goals
+
+- Preserve the original VM's MAC address
+- Preserve the original VM's IP address
+- Specify the gateway IP address of the imported VM so it can keep the same
```

Comment:
Do we know if the ip address is at the same subnet ?

Reply:
Yeah, we'll only take care of that scenario. 

I'll add it to the list of non-goals.

Reply:
Also put at the non goals adding non default gw routes.